packages/dashboard/src/app/views/myStories/filters/taxonomy/useTaxonomyFilters.js

```diff
@@ -0,0 +1,152 @@
+
+/**
+ * Hook used for taxonomy filters logic.
+ * Initializes the taxonomy filters data.
+ *
+ * @return {Array} [taxonomies, queryTaxonomyTerms] initial taxonomy filters data and a function to query the terms.
+ */
+
+function useTaxonomyFilters() {
+  const [taxonomies, setTaxonomies] = useState([]);
+
+  const { getTaxonomies, getTaxonomyTerms } = useApi(
+    ({
+      actions: {
+        taxonomyApi: { getTaxonomies, getTaxonomyTerms },
+      },
+    }) => ({
+      getTaxonomies,
+      getTaxonomyTerms,
+    })
+  );
+
+  /**
+   * Query individual taxonomy data.
+   * This is needed to initialize the primaryOptions and needed to search and set queriedOptions.
+   *
+   * @param {Object} taxonomy object holding taxonomy data, including restBase and restPath used to fetch individual taxonomy terms
+   * @param {string} search string use to query taxonomy by name
+   * @return {Array} taxonomy terms
+   */
+  const queryTaxonomyTerms = useCallback(
+    async (taxonomy, search) => {
+      const { restBase, restPath } = taxonomy;
+      const terms = await getTaxonomyTerms(restPath, { search, per_page: -1 });
+      return terms.map((t) => ({
+        ...t,
+        restBase,
+        restPath,
+      }));
+    },
+    [getTaxonomyTerms]
+  );
+
+  /**
+   * Query all the taxonomies.
+   * This should only be needed once get all the taxonomies and set individual term data.
+   *
+   * @see queryTaxonomyTerms
+   */
+  const queryTaxonomies = useCallback(async () => {
+    const data = await getTaxonomies();
+    const promises = [];
+    const hierarchicalTaxonomies = data.filter(({ hierarchical }) =>
+      Boolean(hierarchical)
+    );
+
+    for (const taxonomy of hierarchicalTaxonomies) {
+      // initialize the data with an empty array
+      taxonomy.data = [];
+      promises.push(queryTaxonomyTerms(taxonomy));
+    }
+
+    const fetched = await Promise.all(promises);
+
+    for (const arr of fetched) {
+      // grab the first elements 'taxonomy' to map the array of terms back to the parent taxonomy
+      const key = arr.at(0)?.taxonomy;
+      if (key) {
+        const taxonomy = hierarchicalTaxonomies.find((h) => h.slug === key);
+        if (taxonomy) {
+          taxonomy.data = arr;
+        }
+      }
+    }
+
+    setTaxonomies(hierarchicalTaxonomies);
+  }, [setTaxonomies, getTaxonomies, queryTaxonomyTerms]);
+
+  /**
+   * Sets up the shape of the taxonomy filters data
+   *
+   * @return {Array} taxonomies filter data
+   */
+  const initializeTaxonomyFilters = useCallback(() => {
+    return taxonomies.map((taxonomy) => ({
+      key: taxonomy.restBase,
+      restPath: taxonomy.restPath,
+      labels: taxonomy.labels,
+      filterId: null,
+      primaryOptions: taxonomy.data,
+      queriedOptions: taxonomy.data,
+      query: queryTaxonomyTerms,
+      placeholder: sprintf(
+        /* translators: All Taxonomies label*/
+        '%s',
+        taxonomy.labels.allItems
+      ),
```

Comment:
```suggestion
      placeholder: taxonomy.labels.allItems,
```

There's no need to use `sprintf` here.